src/compiler/crystal/command.cr

```diff
@@ -425,26 +425,30 @@ class Crystal::Command
     end
   end
 
-  private def setup_simple_compiler_options(compiler, opts)
-    opts.on("-d", "--debug", "Add symbolic debug info") do
-      compiler.debug = true
+  private def setup_simple_compiler_options(compiler, opts, no_codegen = false)
+    unless no_codegen
+      opts.on("-d", "--debug", "Add symbolic debug info") do
+        compiler.debug = true
+      end
     end
     opts.on("-D FLAG", "--define FLAG", "Define a compile-time flag") do |flag|
       compiler.flags << flag
     end
     opts.on("--error-trace", "Show full error trace") do
       compiler.show_error_trace = true
     end
-    opts.on("--release", "Compile in release mode") do
-      compiler.release = true
-    end
-    opts.on("-s", "--stats", "Enable statistics output") do
-      compiler.stats = true
-    end
     opts.on("-h", "--help", "Show this message") do
       puts opts
       exit
     end
+    unless no_codegen
+      opts.on("--release", "Compile in release mode") do
+        compiler.release = true
+      end
+      opts.on("-s", "--stats", "Enable statistics output") do
```

Comment:
See [above](https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/command.cr#L354). In usual compiler option, `--stat` dose not appear if it is not code generation command.